### PR TITLE
make repo reuse compliant

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**How to categorize this PR?**

/area compliance
/kind cleanup

**What this PR does / why we need it**:
* Deleted `SECURITY.md` file. This will be managed centrally
* Added `.reuse/dep5` file and declared copied/3rd party source code accordingly
* Added REUSE badge in `README.md`
* Added licenses
* Change modification copyright statements in various files

### CODE REVIEW

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

File: .reuse/dep5 [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# --------------------------------------------------
+# third-party
+
+# --- copied source code ---
+Files:
+  test/envtest/port.go
+  test/framework/cleanup.go
+  hack/cherry-pick-pull.sh
+  hack/tools/bin/darwin-arm64/include/google/protobuf/*.proto
+  hack/tools/bin/darwin-arm64/include/google/protobuf/compiler/*.proto
```
Comment (vpnachev) [inline .reuse/dep5]: There are no such files, can you double check?

Comment (RaphaelVogel) [reply .reuse/dep5]: correct. Those were local files. Thx for spotting. Have removed them

**Pull request merged**